e2e/tests/upgrades/upgrade_test.go

```diff
@@ -1031,6 +1040,251 @@ func (s *UpgradeTestSuite) TestV8ToV8_1ChainUpgrade_ChannelUpgrades() {
 	})
 }
 
+func (s *UpgradeTestSuite) TestV8ToV9ChainUpgrade() {
```

Comment:
not something for today, but it seems like it could be a good idea to parameterize these upgrade tests. It looks like a lot of them are 90% the same.

Reply:
one of the things I've been thinking about a bit is basically extracting re-usable fns from our existing E2Es and making the actual test cases super short, something like

```golang
   PerformTransferV1(chainA, chainB)  
   UpgradeChain(chainA, newVersion)
   PerformTransferV2(chainA, chainB)
```

Think it will be possible to reduce a huge amount of duplication in the E2Es.